element_array_ephys/ephys_acute.py

```diff
@@ -140,57 +154,77 @@ def auto_generate_entries(cls, session_key):
         Method to auto-generate ProbeInsertion entries for a particular session
         Probe information is inferred from the meta data found in the session data directory
         """
-        session_dir = find_full_path(get_ephys_root_data_dir(),
-                                  get_session_directory(session_key))
+        session_dir = find_full_path(
+            get_ephys_root_data_dir(), get_session_directory(session_key)
+        )
         # search session dir and determine acquisition software
-        for ephys_pattern, ephys_acq_type in zip(['*.ap.meta', '*.oebin'],
-                                                 ['SpikeGLX', 'Open Ephys']):
+        for ephys_pattern, ephys_acq_type in zip(
+            ["*.ap.meta", "*.oebin"], ["SpikeGLX", "Open Ephys"]
+        ):
             ephys_meta_filepaths = list(session_dir.rglob(ephys_pattern))
             if ephys_meta_filepaths:
                 acq_software = ephys_acq_type
                 break
         else:
             raise FileNotFoundError(
-                f'Ephys recording data not found!'
-                f' Neither SpikeGLX nor Open Ephys recording files found in: {session_dir}')
+                f"Ephys recording data not found!"
+                f" Neither SpikeGLX nor Open Ephys recording files found in: {session_dir}"
+            )
 
         probe_list, probe_insertion_list = [], []
-        if acq_software == 'SpikeGLX':
+        if acq_software == "SpikeGLX":
             for meta_fp_idx, meta_filepath in enumerate(ephys_meta_filepaths):
                 spikeglx_meta = spikeglx.SpikeGLXMeta(meta_filepath)
 
-                probe_key = {'probe_type': spikeglx_meta.probe_model,
-                             'probe': spikeglx_meta.probe_SN}
-                if (probe_key['probe'] not in [p['probe'] for p in probe_list]
-                        and probe_key not in probe.Probe()):
+                probe_key = {
+                    "probe_type": spikeglx_meta.probe_model,
+                    "probe": spikeglx_meta.probe_SN,
+                }
+                if (
+                    probe_key["probe"] not in [p["probe"] for p in probe_list]
+                    and probe_key not in probe.Probe()
```

Comment:
`probe_key not in probe.Probe()` is a database query. I think it will be more efficient to skip this test and just do add `skip_duplicates=True` in the insert below.